src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/actions/GenerateTestsActionClass.kt

```diff
@@ -22,14 +22,13 @@ class GenerateTestsActionClass : AnAction() {
      * @param e an action event that contains useful information and corresponds to the action invoked by the user
      */
     override fun actionPerformed(e: AnActionEvent) {
-        // Return if EvoSuite is already running
-        val project = e.project ?: return
-        val runnerService = project.service<RunnerService>()
-        if (runnerService.verifyIsRunning()) return
-
         val psiFile: PsiFile = e.dataContext.getData(CommonDataKeys.PSI_FILE) ?: return
         val linesToInvalidateFromCache = calculateLinesToInvalidate(psiFile)
 
+        val project = e.project ?: return
+        val runnerService = project.service<RunnerService>()
+        if (!runnerService.verify(psiFile)) return
```

Comment:
Could this perhaps be moved to `createEvoSuiteRunner`?

Reply:
For now we want to keep it outside, because the Runner should only be responsible for running EvoSuite